Src/VimCore/Interpreter_Parser.fs

```diff
@@ -1548,6 +1548,49 @@ type Parser
 
         result
 
+    /// Parse out a string option value from the token stream.  The string ends
+    /// on whitespace or vertical bar. If the option is a filename, backslash
+    // escapes only non-filename characters.  Otherwise, it escapes all characters.
+    ///
+    /// help option-backslash
+    member x.ParseOptionBackslash isFileName = 
+        use reset = _tokenizer.SetTokenizerFlagsScoped TokenizerFlags.AllowDoubleQuote
+
+        let builder = System.Text.StringBuilder()
+        let mutable isDone = false
+        while not isDone && not _tokenizer.IsAtEndOfLine do
+            match _tokenizer.CurrentChar with
+            | '\\' ->
+
+                // Escape the next character.
+                _tokenizer.MoveNextChar()
+                if _tokenizer.IsAtEndOfLine then
+                    builder.AppendChar '\\'
+                    isDone <- true
+                else
+                    let char = _tokenizer.CurrentChar
+                    if isFileName && CharUtil.IsFileNameChar char then
+                        builder.AppendChar '\\'
+                        builder.AppendChar char
+                    else
+                        builder.AppendChar char
+                    _tokenizer.MoveNextChar()
+
+            | char ->
+                if CharUtil.IsBlank char || "|".Contains(char.ToString()) then

```

Comment:
Why the Contains vs. just `'|' = char`?

Reply:
This mimicked a longer list from the previous code:

```f#
CharUtil.IsLetterOrDigit c || @"-:\.,<>~[]".Contains(c.ToString())
```

So `Contains` here is a placeholder for any non-blank unescaped character that terminates the string.

According to `:help option-backslash`, the only unescaped characters that can terminate an option-backslash string are vertical bar (which I don't think we support elsewhere yet?) and double quote (which starts a comment).

So technically, double quote needs to be added to it, but I don't think anything else does.